src/spi.rs

```diff
@@ -157,7 +160,7 @@ pub struct SpiBusConfig<USCI: SpiUsci, STATE> {
     _phantom: PhantomData<STATE>,
 }
 
-impl<USCI: SpiUsci> SpiBusConfig<USCI, NoClockSet> {
+impl<USCI: SpiUsci> SpiConfig<USCI, NoClockSet> {
     /// Create a new configuration for setting up a EUSCI peripheral in SPI mode
     pub fn new(usci: USCI, mode: Mode, msb_first: bool) -> Self {
```

Comment:
This constructor relies on `Mode` from ehal 1.0, which is technically a distinct type from `Mode` in ehal 0.2. Should we add a separate 0.2-specific constructor or should we just expect 0.2 users to use the 1.0 types?

As an aside, the fact that `Mode` is even part of ehal is strange, since it's not used by any of the traits.